### DISCUSSION TITLE
API to get all cake arguments

### DISCUSSION THREAD

**Discussion by junjiequ** [cake-build/cake#3054] [Q&A] [answered]
Hi communicity

I am looking for an API that could return me all the cake arguments availabe during the runtime. But I am not able to find any. 
The immediate one I could think is `CakeContext.Arguments`, but it seems that the lookup table is not exposed in the `CakeArguments` class.

Could anyone point me the direction here please? 
Thanks in advance!

**Comment by augustoproiete** [comment, marked as answer]
Cake doesn't have this API at the moment [but it's being considered for a future version](https://github.com/cake-build/cake/issues/2690).

Would you be able to provide an example where this would be useful?

One workaround for now would be to use [`Environment.GetCommandLineArgs`](https://docs.microsoft.com/en-us/dotnet/api/system.environment.getcommandlineargs?view=net-5.0) and parse the arguments yourself to build the lookup table you mentioned or, depending on what you need, a string search or regex search might do the trick.

```csharp
var args = System.Environment.GetCommandLineArgs();

foreach (var arg in args)
{
    Information("{0}", arg);
}

// ...
```

**Reply by junjiequ** [reply]
Thanks for the help.

My use case might be quite dumb: I am merging all cake arguments together with the Environment Variables, after that I need to filter the key/value pairs based on certain key prefixes.

I know I could achive it using other ways... 
But since there is an API to get all Environment Variables already, instinctively I am looking for a similar API from cake framework. That's where the question comes from...
